mne/stats/regression.py

```diff
@@ -283,7 +283,7 @@ def solver(X, Y):
     n_samples = len(times)
     cond_length = dict()
     all_n_lags = [int(tmax_s[cond] - tmin_s[cond]) for cond in conds]
-    X = np.empty((n_samples, sum(all_n_lags)), dtype=float)
+    X = np.zeros((n_samples, sum(all_n_lags)), dtype=float)
```

Comment:
using empty assuming X will be init with zeros (cf. below when you np.where(np.any(X, axis=1))[0]) is terribly dangerous. This line is a bug fix.